django/db/backends/sqlite3/base.py

```diff
@@ -476,3 +494,193 @@ def _sqlite_regexp(re_pattern, re_string):
+        warnings.warn(e1)
+    except ValueError as e2:
+        warnings.warn(e2)
+    finally:
+        return res
+
+
+def _sqlite_atan(x):
+    res = None
+    try:
+        res = math.atan(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_atan2(x, y):
+    res = None
+    try:
+        res = math.atan2(x, y)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_ceiling(x):
+    res = None
+    try:
+        res = math.ceil(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_cos(x):
+    res = None
+    try:
+        res = math.cos(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_cot(x):
+    res = None
+    if x == 0:
+        return None
+    try:
+        res = -math.tan(3.0 * math.pi / 2.0 + x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_degrees(x):
+    res = None
+    try:
+        res = math.degrees(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_exp(x):
+    res = None
+    try:
+        res = math.exp(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_floor(x):
+    res = None
+    try:
+        res = math.floor(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_log(x):
+    res = None
+    try:
+        res = math.log(x)
+    except TypeError as e:
+        warnings.warn(e)
+    except ValueError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_mod(x, y):
+    res = None
+    try:
+        res = x % y
+    except TypeError as e:
+        warnings.warn(e)
+    except ZeroDivisionError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_pi():
+    return math.pi
+
+
+def _sqlite_power2(x, y):
+    res = None
+    try:
+        res = x**y
+    except TypeError as e:
+        warnings.warn(e)
+    except ValueError as e:
+        warnings.warn(e)
+    except ZeroDivisionError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_radians(x):
+    res = None
+    try:
+        res = math.radians(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_sin(x):
+    res = None
+    try:
+        res = math.sin(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+
+def _sqlite_sqrt(x):
+    res = None
+    try:
+        res = math.sqrt(x)
+    except TypeError as e1:
+        warnings.warn(e1)
+    except ValueError as e2:
+        warnings.warn(e2)
+    finally:
+        return res
+
+
+def _sqlite_tan(x):
+    res = None
+    try:
+        res = math.tan(x)
+    except TypeError as e:
```

Comment:
You don't have tests for any of these exceptions. I'm not sure if the exceptions should pass through Django and be handled by the database. This is essentially the behavior of the three other backends.